### PR TITLE
nsc-events-nestjs_1_29_createuserdto

### LINKED ISSUE #29: Create User DTO complete with validation
As a backend developer I will setup a User Data Transfer Object.
I will enable Validation by using the decorators provided by nest.js. 

Work closely with @Exochos (Jeremy Ward) as he is working on issue #28 which depends on the User DTO.
Follow the same file structure that is laid out in the auth and activity folder.
Tutorial:
https://www.youtube.com/watch?v=MfSSwS0MRt8&list=PL_cUvD4qzbkw-phjGK2qq0nQiG6gw1cKK&index=7
Documentation:
https://docs.nestjs.com/techniques/validation

> Keep in mind the libraries are already installed and implemented for the activity. You can reference the DTO there or reach out to @theGaryLarson if you get stuck.

Happy Coding!

### PR DESCRIPTION
**Pull request by Ali-7s**
Resolves #29 

This PR adds the CreateUserDto class with validation.

### CODE REVIEW

**Review by Seiyaroo — DISMISSED** [review]
LGTM, no major changes needed :)

**Comment by Exochos** [comment]
lgtm

File: src/user/dto/create-user.dto.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  @IsNotEmpty()
+  @IsString()
+  @IsAlpha()
+  @MinLength(2, { message: 'Name must be longer than 2 letters.' })
+  name: string;
+
+  @IsNotEmpty()
+  @IsString()
+  @MinLength(8, { message: 'Password must contain 8 characters or more.' })
+  password: string;
```
Comment (theGaryLarson) [inline src/user/dto/create-user.dto.ts:30]: Here do not include password. This dto minimally needs informationi for an admin to set an user's role.
Email and password is used in user auth.

**Review by theGaryLarson — CHANGES_REQUESTED** [review]
Overall looks good. Do not include password in this dto. Even though it is encrypted we do not want to expose it unnecessarily.

File: src/user/dto/create-user.dto.ts (hunk 1 above)
Comment (Ali-7s) [reply src/user/dto/create-user.dto.ts]: Removed the password field

**Comment by taylorpapke** [comment]
@theGaryLarson I'm noticing that on PR's where another push is made, CI is running Continuous-Integration / build-and-test and not completing. @Ali-7s I'm pretty sure if this PR is closed and reopened, it will run CI (on pull-request) / Continuous-Integration / build-and-test (pull_request) and complete. Perhaps something to look into more.

**Comment by taylorpapke** [comment]
Sorry, I accidentally closed with comment rather than just commenting!

File: src/user/dto/create-user.dto.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { Role } from '../../auth/schemas/user.schema';
+import {
+  IsAlpha,
```
Comment (Seiyaroo) [inline src/user/dto/create-user.dto.ts:3]: I'm not familiar with the isAlpha, am assuming that means Alphanumeric?

**Review by Seiyaroo — DISMISSED** [review]
Based on the current settings I feel this is approvable but we have some outstanding changes maybe it looks like?

File: src/user/dto/create-user.dto.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  @IsNotEmpty()
+  @IsString()
+  @IsEmail({}, { message: 'Please enter a valid email.' })
+  email: string;
+
+  @IsNotEmpty()
+  @IsString()
+  @IsAlpha()
+  @MinLength(2, { message: 'Name must be longer than 2 letters.' })
+  name: string;
```
Comment (theGaryLarson) [inline src/user/dto/create-user.dto.ts:25]: Must be equal to or longer than 2 letters. The 2 is inclusive.

**Review by theGaryLarson — APPROVED** [review]
Looks good @Ali-7s

**Review by theGaryLarson — CHANGES_REQUESTED** [review]
Just the message on min length of name needs a bit of adjustment. Looks good!

**Review by Seiyaroo — APPROVED** [review]
I believe this was the only standing issue from Gary's request. With that resolved, this LGTM.

**Review by Seiyaroo — APPROVED** [review]
LGTM, I think the only requested change was the "Equal or greater than" change.

**Review by theGaryLarson — DISMISSED** [review]
Looks good

**Pull request merged**